sqlmesh/dbt/target.py

```diff
@@ -672,6 +676,120 @@ def to_sqlmesh(self) -> ConnectionConfig:
         )
 
 
+class TrinoConfig(TargetConfig):
+    """
+    Project connection and operational configuration for the Trino target.
+
+    Args:
+        method: The Trino authentication method to use
+        host: The server host to connect to
+        port: The MSSQL server port to connect to
+        database: Name of the database
+        roles: Trino catalog roles
+        session_properties: Trino session properties
+        retries: Number of times to retry if the Trino connector encounters an error
+        timezone: The timezone to use for the Trino session
+        http_headers: HTTP Headers to send alongside requests to Trino
+        http_scheme: The HTTP scheme to use for requests to Trino (default: http, or https if kerberos, ldap or jwt auth)
+    """
+
+    type: Literal["trino"] = "trino"
+    host: str
+    database: str
+    schema_: str = Field(alias="schema")
+    port: int = 443
+    method: str
+    user: t.Optional[str] = None
+
+    threads: int = 1
+    roles: t.Optional[t.Dict[str, str]] = None
+    session_properties: t.Optional[t.Dict[str, str]] = None
+    retries: int = 3
+    timezone: t.Optional[str] = None
+    http_headers: t.Optional[t.Dict[str, str]] = None
+    http_scheme: t.Optional[str] = None
+    prepared_statements_enabled: bool = True  # not used by SQLMesh
+
+    # ldap authentication
+    password: t.Optional[str] = None
+    impersonation_user: t.Optional[str] = None
+
+    # kerberos authentication
+    keytab: t.Optional[str] = None
+    krb5_config: t.Optional[str] = None
+    principal: t.Optional[str] = None
+    service_name: str = "trino"
+    hostname_override: t.Optional[str] = None
+    mutual_authentication: bool = False
+    force_preemptive: bool = False
+    sanitize_mutual_error_response: bool = True
+    delegate: bool = False
+
+    # jwt authentication
+    jwt_token: t.Optional[str] = None
+
+    # certificate authentication
+    client_certificate: t.Optional[str] = None
+    client_private_key: t.Optional[str] = None
+    cert: t.Optional[str] = None
+
+    def default_incremental_strategy(self, kind: IncrementalKind) -> str:
+        return "append"
+
+    @classproperty
+    def relation_class(cls) -> t.Type[BaseRelation]:
+        from dbt.adapters.trino.relation import TrinoRelation
+
+        return TrinoRelation
+
+    @classproperty
+    def column_class(cls) -> t.Type[Column]:
+        from dbt.adapters.trino.column import TrinoColumn
+
+        return TrinoColumn
+
+    def to_sqlmesh(self) -> ConnectionConfig:
+        method_to_auth_enum = {
```

Comment:
Should this be a class constant?

Reply:
Example for reference: https://github.com/TobikoData/sqlmesh/blob/3e4f25ad31ee7bbdcacb51d68d9f2506187dd00c/sqlmesh/core/config/base.py#L100